### PR TITLE
BUGFIX: cli parameters being misidentified as 'required'

### PR DESCRIPTION
**Pull request by thilgen**
Noticed this while working with the `zendesk-adaptor` after the recent changes to add `token` to the cli options (for zendesk api token usage).

# Discovery

Attempting to add the `zendesk` service to an environment worked okay when using the interactive mode

`node ~/salto/packages/cli/dist/src/main.js account add zendesk`

However, trying to do this as a one-shot via the `--login-parameters` failed

`node ~/salto/packages/cli/dist/src/main.js account add zendesk --login-parameters username=chris token=abc123 subdomain=testdomain`
```
Error: Could not login to zendesk: Missing the following login parameters: password
```
# Root Cause

The comparison operator utilized in `createConfigFromLoginParameters()` (in `packages/cli/src/commands/account.ts`) scans the param field annotations looking for the **required** params using

`field.annotations[CORE_ANNOTATIONS.REQUIRED] !== false)`

rather than 

`field.annotations[CORE_ANNOTATIONS.REQUIRED] === true)`

Which means the current code considers fields with non-empty `annotations` - that do not have `_required` - as **required**

This appears to be an incorrect assumption based on this comment in `/packages/adapter-utils/test/element.test.ts`

```
// For fields that are not required the _required annotation is not mandatory
// so it is possible to have no annotations, annotations without required and required false
```
# Unit Tests

## New Test

Exported`createConfigFromLoginParameters()` and added a new test in `packages/adapter-utils/test/element.test.ts`

```
const credentialsType = {
  elemID: new ElemID('MockAdaptorName'),
  fields: {
    username: {
       annotations: {
      }
    },
    token: {
      annotations: {
        _required: true
      }
    }
  }
}
```

Results of new test: **Failure before change**
```
const loginParameters = ['token=token456']
```
```
account command group › createConfigFromLoginParameters() helper › when login parameters are marked required › should work when the required param is present

    Missing the following login parameters: username
```

Results of new test: **Success after change**
```
const loginParameters = ['token=token456']
const loginParameters = ['username=user123']
const loginParameters = ['password=password789']
```
```
      when login parameters are marked required
        ✓ should work when the required param is present (1 ms)
        ✓ should throw when the required param is missing - with other matches (6 ms)
        ✓ should throw when the required param is missing - with no other matches (2 ms)
```

## Modification of existing mock

A prev unit test was incorrectly depending on this behavior via the following mock

```
export const mockCredentialsType = (adapterName: string): AdapterAuthentication => {
  const configID = new ElemID(adapterName)
  return {
    basic: {
      credentialsType: new ObjectType({
        elemID: configID,
        fields: {
          username: { refType: BuiltinTypes.STRING },
          password: { refType: BuiltinTypes.STRING },
          token: {
            refType: BuiltinTypes.STRING,
            annotations: {},
          },
          sandbox: { refType: BuiltinTypes.BOOLEAN },
        },
      }),
    },
  }
}
```

And the test was assuming if `sandbox` was omitted then the options should be rejected as malformed. However, since no fields in this mock have the `_required` annotation - none of the parameters should be considered **required.**

Have adjusted the mock to annotate `sandbox` as **required** so that the test now works as intended (fails when a properly annotated **required** param is missing.

# Validation
## Build and Test
```
cd packages/cli

yarn format
yarn build
yarn test

Test Suites: 20 passed, 20 total
Tests:       459 passed, 459 total
Snapshots:   0 total
Time:        29.905 s, estimated 32 s
Ran all test suites.
```
## Original Use case
```
node ~/salto/packages/cli/dist/src/main.js account add zendesk --login-parameters username=chris token=abc123 subdomain=testdomain

✔ Login information successfully updated!

✔ zendesk was added to the environment
```

### CODE REVIEW

File: packages/cli/test/commands/account.test.ts [hunk 1]
```diff
@@ -595,6 +593,44 @@ describe('account command group', () => {
...
+            annotations: {
+            }
+          },
+          token: {
+            annotations: {
+              _required: true
+            }
+          }
+        }
+      }
```
Comment (ori-moisis) [inline packages/cli/test/commands/account.test.ts:611]: the input for the function should be an `ObjectType`
```suggestion
      const credentialsType = new ObjectType({
        elemID: new ElemID('MockAdaptorName'),
        fields: {
          username: {
            annotations: {
            }
          },
          token: {
            annotations: {
              _required: true
            }
          }
        }
      })
```

**Review by ori-moisis — COMMENTED** [review]
The fix seems correct.
There are a few issues with the tests.
Also, this failed a format check, please run `yarn format` to fix formatting issues

File: packages/cli/test/commands/account.test.ts [hunk 2]
```diff
@@ -367,19 +378,6 @@ describe('account command group', () => {
           })
         })
         describe('when called with invalid login parameters', () => {
-          it('should fail when called with missing parameter', async () => {
```
[the remark this answers is not in the document]
Comment (thilgen) [reply packages/cli/test/commands/account.test.ts:370]: Okay - I will see about finding a replacement.

[the remark this answers is not in the document]
Comment (thilgen) [reply packages/cli/test/commands/account.test.ts]: Added back test - Adjusted mock to the intended shape

File: packages/cli/test/commands/account.test.ts (hunk 1 above)
Comment (thilgen) [reply packages/cli/test/commands/account.test.ts]: Done.

**Comment by thilgen** [comment]
Updated PR based on comments; Updated Description to align with latest changes; Ready for re-review.

**Review by ori-moisis — APPROVED** [review]

**Pull request merged**